src/main/java/com/tastetracker/domain/restaurant/Restaurant.java

```diff
@@ -22,6 +24,6 @@ public class Restaurant {
     @ManyToOne
     @JoinColumn(name = "address_id", referencedColumnName = "id")
     private Address address;
-    @OneToMany(mappedBy = "restaurant")
-    private Set<RestaurantCategory> restaurantCategory;
+    @ManyToMany
+    private Set<Category> category = new HashSet<>();
```

Comment:
zgodnie z tym że jest tabela pośrednicząca to też tutaj powinno być to ujęte czyli dodać :
@JoinTable(
            name = "restaurant_category",
            joinColumns = @JoinColumn(name = "restaurant_id", referencedColumnName = "id"),
            inverseJoinColumns = @JoinColumn(name = "category_id", referencedColumnName = "id")
    )